### PR TITLE
Tensors

### PR DESCRIPTION
**Pull request by lcnbr**
Adds purely numerical tensors, mixed symbolic numeric tensors and symbolic tensor implementations.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -15,6 +15,9 @@ repository = "https://github.com/alphal00p/gammaloop"
 inherits = "dev"
 opt-level = 2
 
+[profile.release]
+#lto=true
```
Comment (ValentinHirschi) [inline Cargo.toml:19]: @benruijl @lcnbr Should we enable lto for release?

Comment (lcnbr) [reply Cargo.toml:19]: Ah I was just messing around to see if it improved performance.

Comment (benruijl) [reply Cargo.toml:19]: Yes, LTO may improve performance. `LTO="thin"` can also be tried

Comment (lcnbr) [reply Cargo.toml:19]: I didn't benchmark it, but just looking at the gamma chain timing, it seems to make it slightly faster. Maybe 10% even. It does make compilation time go through the roof. "Thin" would be a middle ground.

**Pull request closed without merging**